src/vm/reflectioninvocation.cpp

```diff
@@ -1381,7 +1386,31 @@ FCIMPL5(Object*, RuntimeMethodHandle::InvokeMethod,
             COMPlusThrow(kNullReferenceException, IDS_INVOKE_NULLREF_RETURNED);
         }
 
-        gc.retVal = InvokeUtil::CreateObjectAfterInvoke(refReturnTargetTH, pReturnedReference);
+        if (!fPropertyGetterCalledThroughSetValue)
+        {
+            gc.retVal = InvokeUtil::CreateObjectAfterInvoke(refReturnTargetTH, pReturnedReference);
+        }
+        else
+        {
+            DWORD cbValueTypeSize;
+            if (refReturnTargetTH.IsValueType() && !refReturnTargetTH.AsMethodTable()->ContainsPointers() && (cbValueTypeSize = refReturnTargetTH.AsMethodTable()->GetNumInstanceFieldBytes()) < sizeof(ARG_SLOT))
```

Comment:
Not sure if the ContainsPointer and IsValueType checks are needed. I didn't want to use "memcpyNoGc" on things with OBJECTREFs in them.